packages/cysync-core/src/pages/OnBoarding/CardAuthentication/index.tsx

```diff
@@ -14,26 +16,46 @@ import { CardTap } from './Dialogs/CardTap';
 import { OnboardingPageLayout } from '../OnboardingPageLayout';
 
 export const CardAuthentication: React.FC = () => {
+  const query = useQuery();
   const lang = useAppSelector(selectLanguage);
 
-  // number of card taps needed for authentication 3 taps per card for 4 cards
+  const totalCards = 4;
+
+  const [tapsPerCard, setTapsPerCard] = useState(3);
+  // total number of card taps needed for authentication for all cards
   const [cardTapState, setCardTapState, isFinalCardTapState] =
-    useStateWithFinality(0, 12);
+    useStateWithFinality(0, tapsPerCard * totalCards);
   const navigateTo = useNavigateTo();
 
-  useWhenDeviceConnected();
+  const cardAuth: OnConnectCallback = async ({ connection, connectDevice }) => {
+    if (!connection) return;
 
-  // replace this with cardAuth function
-  addKeyboardEvents({
-    ' ': () => {
-      setCardTapState(s => s + 1);
-    },
-  });
+    const app = await ManagerApp.create(await connectDevice(connection.device));
+    for (let cardNumber = 1; cardNumber <= totalCards; cardNumber += 1) {
+      await app.authCard({
+        cardNumber,
+        isPairRequired: tapsPerCard === 3,
+        onEvent: flowStatus => {
+          const newState =
+            Math.min(flowStatus - 1, tapsPerCard) +
```

Comment:
Link this algorithm with the Status enum, such that if it is modified, we won't need to come back here and change it.
```suggestion
            Math.min(
              flowStatus - AuthCardStatus.AUTH_CARD_STATUS_SERIAL_SIGNED + 1,
              tapsPerCard,
            ) +
```